pkg/webhook/webhook.go

```diff
@@ -386,3 +391,22 @@ func genCertAndKeyFile(certData, keyData []byte, certDir string) error {
 	klog.V(2).InfoS("successfully generate certificate and key files")
 	return nil
 }
+
+// bindWebhookConfigToFleetSystem sets the OwnerReference of the webhook config to a cluster scoped Fleet resource.
+func bindWebhookConfigToFleetSystem(ctx context.Context, k8Client client.Client, validatingWebhookConfig *admv1.ValidatingWebhookConfiguration) error {
+	var fleetNs corev1.Namespace
+	if err := k8Client.Get(ctx, client.ObjectKey{Name: "fleet-system"}, &fleetNs); err != nil {
+		return err
+	}
+
+	ownerRef := metav1.OwnerReference{
+		APIVersion:         fleetNs.GroupVersionKind().GroupVersion().String(),
+		Kind:               fleetNs.Kind,
+		Name:               fleetNs.GetName(),
+		UID:                fleetNs.GetUID(),
+		BlockOwnerDeletion: pointer.Bool(false),
+	}
+
+	validatingWebhookConfig.OwnerReferences = []metav1.OwnerReference{ownerRef}
```

Comment:
Do we know for a fact that validatingWebHook doesn't have any other owner references ?

Reply:
Yes, this set of routines are what construct and create the config within the cluster. The config must also not exist. There is no current concern for existing owner references. 

https://github.com/Azure/fleet/blob/8e4a6cb7dfb57916fa52ffb39308fbdf6263512e/pkg/webhook/webhook.go#L184-L191